spark/dl/src/main/scala/com/intel/analytics/bigdl/parameters/AllReduceParameterManager.scala

```diff
@@ -0,0 +1,333 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.intel.analytics.bigdl.parameters
+
+import java.util.concurrent.atomic.AtomicInteger
+import java.util.concurrent.{Callable, Executors, Future, ThreadFactory}
+
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import com.intel.analytics.bigdl.utils.{Engine, T, Table}
+import org.apache.commons.lang.exception.ExceptionUtils
+import org.apache.log4j.Logger
+import org.apache.spark.sparkExtension.{ParameterManagerMaster, SparkExtension}
+import org.apache.spark.SparkEnv
+import org.apache.spark.storage.{BlockId, BlockManagerWrapper, StorageLevel}
+
+import scala.collection.JavaConverters._
+import scala.collection.mutable.HashMap
+import scala.reflect._
+
+object AllReduceParameterManager {
+  val logger = Logger.getLogger(getClass)
+
+  private val nextId = new AtomicInteger(0)
+
+  private val pm = new HashMap[String, AllReduceParameterManager]()
+
+  private var executorIdMap: HashMap[String, Int] = null
+
+  def setExecutorMap(map: HashMap[String, Int]): Unit = {
+    executorIdMap = map
+  }
+
+  def get(id: Int, executorId: String): AllReduceParameterManager = {
+    val eid = id + "exe" + executorIdMap(executorId)
+    if (pm.contains(eid)) pm(eid)
+    else null
+  }
+
+  def createParameterManager[T: ClassTag](executorId: Int, executorNum: Int, partitionNum: Int,
+    size: Int, port: Int = -1): AllReduceParameterManager = {
+    val id = nextId.getAndIncrement()
+    val conf = SparkEnv.get.conf
+    val master = ParameterManagerMaster.createEnv(conf, port)
+    val p = new AllReduceParameterManager(id, executorId, executorNum, partitionNum, size, master)
+    pm.put(id + "exe" + executorId, p)
+    p
+  }
+}
+
+/**
+ * Represent parameters stored on the block manager. In distributed optimization, we put parameters
+ * on block manager of spark. Each worker syncs parameters through the block manager. Block manager
+ * here serves as a parameter server.
+ *
+ * A Tensor is sliced into `executorNum` chunks and each chunk is assigned to a particular node
```

Comment:
Yes and agreed, it's on plan. But since it's not a small change, i would like to do it in another pr.